euphonic/design/05_powder_averaging_discussion.md

```diff
@@ -0,0 +1,318 @@
+      sample S(**q**,ω) in a response formalism from dynamic data.
+    - The latter approach maps better to PACE; at each **q** we go to
+      the force constants and get all the active ω. This has the
+      advantage that every sample should find some signal, without the
+      risk of hitting "empty space".
+
+
+<a id="orgd1b667c"></a>
+
+### Spherical sampling
+
+-   SpinW seems to use a Fibonacci-related method for uniform
+    sampling at spherical surfaces
+    -   points per shell has to be a Fibonacci number
+        - That's not a deal-breaker but could be annoying when exploring convergence
+    -   or can sample *randomly* at *each* spherical shell surface
+-   MDANSE requires the user to set up a q lattice for sampling,
+    including a "spherical grid" option with a set range of "shells"
+    and "vectors".
+-   Another approach that doesn't seem to be in use but is perhaps
+    a truer estimate of the integral is to sample _within a q bin_
+    rather than exactly at the surface.
+
+
+<a id="org93896a1"></a>
+
+### Unknown
+
+-   In OCLIMAX, TASK=1 allegedly performs a "full powder
+    calculation" with coherent effects (as opposed to analytic
+    inhoherent method). This is not affected by the user q-bin
+    options, and neither manual nor paper gives any information
+    about the sampling approach.
+
+
+<a id="orgf40d9b1"></a>
+
+# Further questions
+
+-   There are two different possible objectives here: uniform sampling
+    **at each q** and uniform sampling **over the spherical
+    volume**. Which is more important?
+
+-   Why do we pre-determine the mod(q) bins?
+    -   easier to compare with instrument
+    -   able to check convergence *at each bin* independently
+    -   in Abins, the simulated *q* bins do not match the TOSCA
+        instrument measurements, in which the bin size increases with
+        *q*.
+
+-   Do we need to include a resolution function before/during powder
+    averaging?
+    -   Different instrument models need to apply resolution functions at different
+        stages; it depends how the function was developed.
+        -   As long as the powder-averaging functions accept/return
+            appropriate Euphonic data types, we shouldn't need to make
+            special allowances at this stage. This can be experimented
+            with if/when new resolution function models are developed.
+
+    -   Ideally one would build a combined output from every simulated pixel
+        of the instrument. This doesn't seem feasible for instruments
+        with thousands of detectors (e.g. MERLIN)
+        -   This approach seems to map to the McVine package's goals
+            better than Horace/Euphonic.
+        -   McVine accounts for instrument design, but takes S(q,ω) as
+            input; no support for 4-D data, which could be needed for such a model.
+        -   The McVine package is somewhat intimidating; such work is
+            not to be taken lightly.
```

Comment:
[McStas](http://mcstas.org/) is also another package which can be used to do something similar: https://www.youtube.com/watch?v=ZcqermI64VA but if anything it can be even more complicated than McVine.